File: app/src/main/java/com/aquamorph/frcmanager/models/Event.java
package com.aquamorph.frcmanager.models;

import java.text.SimpleDateFormat;
import java.util.Date;
import java.util.Locale;

/**
 * Stores information about an event.
 *
 * @author Christian Colglazier
 * @version 3/29/2016
 */
public class Event implements Comparable {

	public String key;
	public String name;
//	public String event_code;
//	public int event_type;
//	public DistrictList district;
//	public String city;
//	public String state_prov;
//	public String country;
	public String start_date;
//	public String end_date;
	public int year;
	public String short_name;
//	public String event_type_string;
//	public int week;
//	public String address;
//	public String postal_code;
//	public String gmaps_place_id;
//	public String gmaps_url;
//	public double lat;
//	public double lng;
//	public String location_name;
//	public String timezone;
//	public String website;
//	public String first_event_id;
//	public String first_event_code;
//	public Webcast[] webcasts;
//	public String[] division_keys;
//	public int playoff_type;
//	public String playoff_type_string;

	@Override
	public int compareTo(Object another) {
		try {

			Date first = new SimpleDateFormat("yyyy-MM-dd", Locale.ENGLISH).parse(this.start_date);
			Date second = new SimpleDateFormat("yyyy-MM-dd", Locale.ENGLISH).parse(((Event) another).start_date);
			return first.compareTo(second);
		} catch (Exception e) {
			e.printStackTrace();
		}
		return 0;
	}
}

File: app/src/main/kotlin/com/aquamorph/frcmanager/models/Event.kt
package com.aquamorph.frcmanager.models

import com.google.gson.annotations.Expose
import com.google.gson.annotations.SerializedName
import java.text.SimpleDateFormat
import java.util.Locale

/**
 * Stores information about an event.
 *
 * @author Christian Colglazier
 * @version 1/31/2020
 */
data class Event(
    @Expose
    @SerializedName("key")
    var key: String,
    @Expose
    @SerializedName("name")
    var name: String,
    @Expose
    @SerializedName("event_code")
    var eventCode: String,
    @Expose
    @SerializedName("event_type")
    var eventType: Int,
    @Expose
    @SerializedName("district")
    var district: DistrictList,
    @Expose
    @SerializedName("city")
    var city: String,
    @Expose
    @SerializedName("state_prov")
    var stateProv: String,
    @Expose
    @SerializedName("country")
    var country: String,
    @Expose
    @SerializedName("start_date")
    var startDate: String,
    @Expose
    @SerializedName("end_date")
    var endDate: String,
    @Expose
    @SerializedName("year")
    var year: Int,
    @Expose
    @SerializedName("short_name")
    var shortName: String,
    @Expose
    @SerializedName("event_type_string")
    var eventTypeString: String,
    @Expose
    @SerializedName("week")
    var week: Int,
    @Expose
    @SerializedName("address")
    var address: String,
    @Expose
    @SerializedName("postal_code")
    var postalCode: String,
    @Expose
    @SerializedName("gmaps_place_id")
    var gmapsPlaceId: String,
    @Expose
    @SerializedName("gmaps_url")
    var gmapsUrl: String,
    @Expose
    @SerializedName("lat")
    var lat: Double,
    @Expose
    @SerializedName("lng")
    var lng: Double,
    @Expose
    @SerializedName("location_name")
    var locationName: String,
    @Expose
    @SerializedName("timezone")
    var timezone: String,
    @Expose
    @SerializedName("website")
    var website: String,
    @Expose
    @SerializedName("first_event_id")
    var firstEventId: String,
    @Expose
    @SerializedName("first_event_code")
    var firstEventCode: String,
//    @Expose
//    @SerializedName("webcasts")
//    var webcasts: Array<WebCast>?
    @Expose
    @SerializedName("division_keys")
    var divisionKeys: ArrayList<String>,
    @Expose
    @SerializedName("playoff_type")
    var playoffType: Int,
    @Expose
    @SerializedName("playoff_type_string")
    var playoffTypeString: String
) : Comparable<Event> {
    override operator fun compareTo(other: Event): Int {
        try {
            return SimpleDateFormat("yyyy-MM-dd", Locale.ENGLISH).parse(this.startDate)!!
                    .compareTo(SimpleDateFormat("yyyy-MM-dd", Locale.ENGLISH).parse(other.startDate))
        } catch (e: Exception) {
            e.printStackTrace()
        }
        return 0
    }
}
